commown_shipping/tests/test_crm_lead.py

```diff
@@ -182,6 +182,24 @@ def test_onchange_expedition_ref(self):
         # Check results
         self.assertEqual(self.lead.expedition_ref, "AAJJPP")
 
+    def test_check_expedition_ref(self):
+        stage = self.env["crm.stage"].create({"name": "TEST [log: check exp-ref]"})
+        # Check Pre-requisite
+        self.assertFalse(self.lead.expedition_ref)
+
+        expected_msg = "Lead has no expedition ref. Please fill it in."
+
+        with self.assertRaises(ValidationError):
```

Comment:
Qu'est-ce qui raise ValidationError ici ? Au minimum à commenter, mais bad smell.

Reply:
C'est la non validation de la contrainte (on utilise un api.constrains) ! Il y a la même pour [stock: check-has-picking] !

Reply:
Mais noté pour le commentaire

Reply:
Ah oui, mais il faut raise ValidationError dans une api.constrains, pas UserError, alors on peut se contenter de : 
```
        with self.assertRaises(ValidationError) as err:
            self.lead.stage_id = stage
        self.assertEqual(err.exception.name, expected_msg)
```

Reply:
Effectivement ! Faut que je le change dans commown_devices aussi alors !